### ISSUE TITLE
Throw the version error only when the previous submitted version is approved 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#19964]
For https://github.com/mozilla/addons-server/issues/19849#issuecomment-1311858536

The error that a version must be higher -> is displayed even if the submitted version has not been approved yet (it is in awaiting review) 

Expected result:
An error is thrown only if the previous version is approved.

**Comment by wagnerand** [comment]
This opens a dangerous gap for versions that were previously public but are now disabled. We should probably check against signed versions regardless of their status.

**Comment by eviljeff** [comment]
> This opens a dangerous gap for versions that were previously public but are now disabled. We should probably check against signed versions regardless of their status.

Agreed - https://github.com/mozilla/addons-server/issues/19788 (the original issue in this maze of follow-ups and reverts) specifically mentioned the current listed version though.  I'm reading this as we want to compare to the latest signed listed version instead?

**Comment by ioanarusiczki** [comment]
@eviljeff 

I verified this on AMO Dev with all addon types for MV2:

- when a lower listed version is submitted while the fist listed version is in awaiting review: submission is possible
- when the first listed version is approved, the next listed version must be higher 
- when the first listed version is approved, then disabled by developers, rejected by reviewers or blocked, the next submitted version must be higher -> the error message is displayed
- when the first version is disabled, rejected or blocked -> a new version with a lower version number can be submitted

From what's been discussed above, that results look good to me. 

I've also tested around with a MV3 sample, seems to be alright. 

On the new API and signing API I do not have time to test it in detail as I did for dev hub but I tried with some addons and it seems to be functioning as expected.
